test/e2e/controller_test.go

```diff
@@ -65,7 +65,7 @@ func loadClusterConfig(t *testing.T, clusterName logicalcluster.Name) *rest.Conf
 	if err != nil {
 		t.Fatalf("failed to load *rest.Config: %v", err)
 	}
-	return rest.AddUserAgent(kcpclienthelper.ConfigWithCluster(restConfig, clusterName), t.Name())
+	return rest.AddUserAgent(kcpclienthelper.SetCluster(restConfig, clusterName), t.Name())
```

Comment:
`SetCluster` does not copy the rest config and modify the host anymore which `ConfigWithCluster` used to do. So we need to do it here:
```suggestion
	return rest.AddUserAgent(kcpclienthelper.SetCluster(rest.CopyConfig(restConfig), clusterName), t.Name())
```

Reply:
@varshaprasad96 thanks for pointing that out. I should have read the small print. I have applied your improvement.